### PR TITLE
Edit_page => was cleaned code, aded avatar for activity, maked applicants List

### PR DESCRIPTION
**Pull request by aghermanHP**
Edit_page => was cleaned code, aded avatar for activity, maked applicants List @nlaptedulce  please merge this pull request

### CODE REVIEW

File: src/assets/main.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
 @import '~bootstrap-vue/dist/bootstrap-vue.css';
 @import url('https://fonts.googleapis.com/css?family=Lato:400,700');
 @import url('https://use.fontawesome.com/releases/v5.3.1/css/all.css');
-
+@import './styles/vueDataTime.scss'; //style for datetime piker
```
Comment (avolchivschi) [inline src/assets/main.scss]: Remove comments

Comment (aghermanHP) [reply src/assets/main.scss]: removed

File: src/components/activity/VActivity.vue [hunk 2]
```diff
@@ -7,11 +7,11 @@
...
           <div class="col-md-9">
             <div class="row">
-              <b-card-title class="col-md-8 mx-auto row">
+              <b-card-title v-bind="Activity.name" class="col-md-8 mx-auto row">
                 {{ Activity.name }}
               </b-card-title>
               
-              <b-card-text class="col-md-4 text-right">
+              <b-card-text class="col-md-4 text-right" v-bind="Activity.owner.name + Activity.owner.surname">
                 {{ Activity.owner.name +' ' + Activity.owner.surname }}
```
Comment (avolchivschi) [inline src/components/activity/VActivity.vue]: Space between + and '

File: src/components/activity/ActivityList.vue [hunk 3]
```diff
@@ -128,7 +153,7 @@
 
 <script>
   import ActivityService from '../../services/activityApi';
-  import { mapState, mapGetters } from 'vuex';
+  import {mapState, mapGetters} from 'vuex';
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: space after { and before }

File: src/components/activity/ActivityList.vue [hunk 4]
```diff
@@ -139,31 +164,110 @@
...
-            console.log(error);
+          .catch (error => {
           });
       },
       redirectToActivityDetails(id) {
-        this.$router.push({ name: 'activity', params: { activityId: id } });
-      },
-    },
-  };
+        this.$router.push ({name: 'activity', params: {activityId: id}});
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: you don't need here a space before (

File: src/components/activity/ActivityList.vue [hunk 5]
```diff
@@ -139,31 +164,110 @@
...
+        }
+      }
     },
-    computed:{
-      ...mapState('account',['user']),
-      ...mapGetters('account', ['userId']),
+    computed: {
+      ...mapState('account', ['user']),
     },
     mounted() {
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: ### Component/instance options order
- name
- inject
- components
- props
- data
- created
- mounted
- computed
- methods
- destroyed

File: src/components/activity/ApplicantsList.vue [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+          height: 0
+        }
+      }
+    },
+    mounted () {
+      this.getListApplicants ()
+    },
+    methods: {
+      getListApplicants () {
+        ActivityService.getApplicantsList (this.$route.params.idActivity).then ((response) => {
```
Comment (nlaptedulce) [inline src/components/activity/ApplicantsList.vue]: you don't need her space before (

File: src/components/activity/ApplicantsList.vue [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+          results: []
+        },
+        window: {
+          width: 0,
+          height: 0
+        }
+      }
+    },
+    mounted () {
+      this.getListApplicants ()
```
Comment (nlaptedulce) [inline src/components/activity/ApplicantsList.vue]: and here

File: src/components/activity/CreateActivity.vue [hunk 8]
```diff
@@ -2,123 +2,110 @@
...
-                  </b-form-select>
-
-                  <span v-if="errors.has('technologies')" class="invalid-feedback">{{ errors.first('Technologies') }}</span>
-                </b-form-group>
-
-                <b-form-group class="col-md-6">
-                  <b-form-select
-                    v-model="typeId"
+                
+                <b-form-group
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue]: you can write in one line this

File: src/components/activity/CreateActivity.vue [hunk 9]
```diff
@@ -152,28 +145,41 @@
...
-        ]
-      };
+        timeStartApplication: '',
+        timeStartDeadline: ''
+      }
+    },
+    created () {
+      let startTime = new Date ();
+
+      startTime.setDate (startTime.getDate () + 1);
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue]: remove space before (

Comment (nlaptedulce) [reply src/components/activity/CreateActivity.vue]: for all methode

File: src/components/activity/CreateActivity.vue [hunk 10]
```diff
@@ -152,28 +145,41 @@
...
     },
     methods: {
+      setDeadline () {
+        let deadline = moment (this.form.application_deadline).add (1, 'days');
+
+        this.form.final_deadline = moment (deadline).toISOString ();
+        this.timeStartDeadline = moment (deadline).toISOString ();
+      },
       createActivity () {
+
```
Comment (nlaptedulce) [inline src/components/activity/CreateActivity.vue]: remove this empty line

File: src/components/activity/Technologies.vue [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  import UserApi from '@/services/userDetailsApi';
+  
+  export default {
+    model: {
+      prop: "technologies",
+      event: "changeTechnologiesList"
+    },
+    props: {technologies: Array},
+    data() {
+      return{
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: space after return

File: src/components/activity/Technologies.vue [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        formTechnologies: [],
+        selectedTechnologies: []
+      }
+    },
+   created(){ this.selectedTechnologies = this.technologies},
+    mounted() {
+      this.selectedTechnologies = this.technologies;
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: maybe it this.selectedTechnologies = this.technologies; should be just in mounted?

File: src/components/activity/Technologies.vue [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+      UserApi.getTechnologies ().then ((response) => {
+        this.formTechnologies = response.data;
+      }).catch (error => {
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: remove catch if you don't use it

File: src/components/activity/Technologies.vue [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+      });
+    },
+    methods: {
+      updateTechnologies (technologies) {
+        this.$emit('changeTechnologiesList', technologies);
+      }
+    }
+  };
+</script>
+<style>
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: remove stype

File: src/components/activity/Types.vue [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+      return {
+        formTypes: [],
+        selectedTypes: []
+      };
+    },
+    mounted () {
+      this.selectedTypes = this.types;
+      ActivityService.getTypes().then((response) =>{
+        this.formTypes = response.data;
+      }).catch(error =>{
```
Comment (nlaptedulce) [inline src/components/activity/Types.vue]: same

File: src/components/activity/VActivity.vue [hunk 16]
```diff
@@ -1,158 +1,245 @@
...
+              <div class="row">
                 <b-card-title class="col-md-8 mx-auto row">
                   {{ activity.name }}
                 </b-card-title>
-
                 <b-card-text class="col-md-4 text-right text-capitalize">
-                  <img v-b-tooltip.hover :title="activity.owner.name + ' ' + activity.owner.surname"
-                       src="../../assets/images/user-image.png" class="user-image" alt="avatar"
-                  >
+                  {{ activity.owner.name+' '+activity.owner.surname }}
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: space after and befor +

File: src/components/activity/VActivity.vue [hunk 17]
```diff
@@ -1,158 +1,245 @@
...
-    </div>
+      </b-container>
+    </section>
   </div>
 </template>
 
 <script>
   import ActivityService from '../../services/activityApi';
-  import { mapState, mapGetters } from 'vuex';
+  import {mapState, mapGetters} from 'vuex';
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue]: same

File: src/components/activity/VActivity.vue [hunk 18]
```diff
@@ -1,158 +1,245 @@
...
+          owner: {
+            avatar: {}
+          }
         },
-      };
+        delete: false,
+        avatarActivityUrl: null
+      }
     },
+
```
Comment (nlaptedulce) [inline src/components/activity/VActivity.vue:149]: remove line
### Component/instance options order

- name
- inject
- components
- props
- data
- created
- mounted
- computed
- methods
- destroyed

File: src/services/activityApi.js [hunk 19]
```diff
@@ -14,11 +14,22 @@ export default {
   createActivity(data){
     return API.post(userEndpoints.ACTIVITIES + '/create', data)
   },
-
+  editActivity(id, data){
+    return API.post( userEndpoints.ACTIVITIES +'/' + id + '/edit', data)
+  },
+  getTypes() {
+    return API.get( 'http://api.pentabee.local/api/activity-types')
```
Comment (nlaptedulce) [inline src/services/activityApi.js]: ???

**Comment by aghermanHP** [comment]
@nlaptedulce  please verify if it's ok

File: src/views/TheHeader.vue [hunk 20]
```diff
@@ -67,34 +67,32 @@
...
-        console.log(error);
-      });
+      })
     },
     methods: {
       onclick () {
-        window.localStorage.removeItem('token');
-        this.$toast.open({
+        window.localStorage.removeItem ('token');
+        this.$toast.open ({
```
Comment (nlaptedulce) [inline src/views/TheHeader.vue]: space

File: src/views/TheHeader.vue [hunk 21]
```diff
@@ -67,34 +67,32 @@
...
       form: {}
     }),
     computed: {
-      ...mapGetters('account',['userId']),
+      ...mapGetters ('account', ['userId']),
     },
-    mounted() {
-      UserApi.userInfo(this.userId).then((response) => {
+    mounted () {
+      UserApi.userInfo (this.userId).then ((response) => {
```
Comment (nlaptedulce) [inline src/views/TheHeader.vue]: space

File: src/views/TheHeader.vue [hunk 22]
```diff
@@ -67,34 +67,32 @@
...
 <script>
   import UserApi from '@/services/userDetailsApi';
-  import { mapGetters } from 'vuex';
+  import {mapGetters} from 'vuex';
 
   export default {
     data: () => ({
       form: {}
     }),
     computed: {
```
Comment (nlaptedulce) [inline src/views/TheHeader.vue:76]: ### Component/instance options order

- name
- inject
- components
- props
- data
- created
- mounted
- computed
- methods
- destroyed

File: src/components/activity/ActivityList.vue [hunk 23]
```diff
@@ -2,113 +2,135 @@
...
+              <div class="row">
+                <div
+                  v-for="result in results"
+                  :key="result.id"
+                  class="col-md-4 row-eq-height"
+                  @click="redirectToActivityDetails(result.id)"
+                >
+                  <b-card bg-variant="light" border-variant="warning" class="cards col-md-12 mt-3">
+                    <div class="row">
+                      <img src="../../assets/images/combs.jpg" class=" activity-image mr-2">
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue]: space

File: src/components/activity/ActivityList.vue [hunk 24]
```diff
@@ -2,113 +2,135 @@
...
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue:52]: space

File: src/components/activity/ActivityList.vue [hunk 25]
```diff
@@ -2,113 +2,135 @@
...
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue:83]: !

File: src/components/activity/ActivityList.vue [hunk 26]
```diff
@@ -2,113 +2,135 @@
...
```
Comment (nlaptedulce) [inline src/components/activity/ActivityList.vue:114]: all img tag in class are space after "

File: src/components/activity/EditActivity.vue [hunk 27]
```diff
@@ -0,0 +1,233 @@
...
+                    class="form-control"
+                    :class="{ 'is-invalid': errors.has('name') }"
+                    @change="edited = true"
+                  />
+                  <span v-if="errors.has('name')" class="invalid-feedback">
+                    {{ errors.first('name') }}
+                  </span>
+                </b-form-group>
+
+                <b-form-group class=" col-md-6">
```
Comment (nlaptedulce) [inline src/components/activity/EditActivity.vue:36]: space

File: src/components/activity/EditActivity.vue [hunk 28]
```diff
@@ -0,0 +1,233 @@
...
+                :delay="100"
+                :adaptive="true"
+              >
+                <div class="example-modal-content text-center mt-5">
+                  <h6>Do you want to save changes for activity?</h6>
+                </div>
+                
+                <div class="row mt-lg-5 ml-3">
+                  <b-button class="col-md-5" variant="dark" @click="cancel">
+                    Cancel</b-button>
```
Comment (nlaptedulce) [inline src/components/activity/EditActivity.vue:144]: or in one line or </b-button> in new line

File: src/components/activity/EditActivity.vue [hunk 29]
```diff
@@ -0,0 +1,233 @@
...
+        technologies: [{ id: null }],
+        types: []
+      },
+      edited: false,
+      technologiesList: []
+    }
+  },
+  
+  computed: {
+    ...mapState('account', ['user']),
```
Comment (nlaptedulce) [inline src/components/activity/EditActivity.vue]: ### Component/instance options order

- name
- inject
- components
- props
- data
- created
- mounted
- computed
- methods
- destroyed

File: src/components/activity/Technologies.vue [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+    this.selectedTechnologies = this.technologies;
+    UserApi.getTechnologies()
+      .then(response => {
+        this.formTechnologies = response.data
+      })
+  },
+  methods: {
+    updateTechnologies(technologies) {
+      this.$emit('changeTechnologiesList', technologies)
+    },
```
Comment (nlaptedulce) [inline src/components/activity/Technologies.vue]: commas

File: src/route/router.js [hunk 31]
```diff
@@ -72,15 +74,31 @@ const router = new Router({
         requiresAuth: true,
       },
     },
+    {
+      path: '/activity/:activityEditId/edit',
+      name: 'activityEdit',
+      component: ActivityEdit,
+      meta: {
+        requiresAuth: true,
```
Comment (nlaptedulce) [inline src/route/router.js]: remove comma

File: src/route/router.js [hunk 32]
```diff
@@ -72,15 +74,31 @@ const router = new Router({
...
+      }
+    },
     {
       path: '/activity-create',
       name: 'createActivity',
       component: CreateActivity,
       meta: {
         requiresAuth: true,
```
Comment (nlaptedulce) [inline src/route/router.js]: and here

File: src/route/router.js [hunk 33]
```diff
@@ -72,15 +74,31 @@ const router = new Router({
...
-      },
+      }
     },
-  ],
+    {
+      path: '/activity/:idActivity/applicants',
+      name: 'applicantsList',
+      component: ApplicantsList,
+      meta: {
+        requiresAuth: true,
```
Comment (nlaptedulce) [inline src/route/router.js]: !

File: src/views/TheHeader.vue (hunk 22 above)
Comment (nlaptedulce) [reply src/views/TheHeader.vue:76]: mai intii mounted apoi computed

File: src/components/activity/ApplicantsList.vue [hunk 34]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  methods: {
+    getListApplicants() {
+      ActivityService.getApplicantsList(this.$route.params.idActivity).then(
+        response => {
+          this.form.results = response.data.results;
+          this.isBusy = false;
+        }
+      );
+    },
```
Comment (nlaptedulce) [inline src/components/activity/ApplicantsList.vue]: commas

File: src/components/activity/ActivityList.vue (hunk 24 above)
Comment (nlaptedulce) [reply src/components/activity/ActivityList.vue:52]: ?

**Pull request merged**